package/gluon-web-admin/luasrc/lib/gluon/config-mode/model/admin/remote.lua

```diff
@@ -76,29 +76,12 @@ function pw2.cfgvalue()
 end
 
 local function set_password(password)
-	local inr, inw = unistd.pipe()
-	local pid = unistd.fork()
-
-	if pid < 0 then
-		return false
-	elseif pid == 0 then
-		unistd.close(inw)
-
-		local null = fcntl.open('/dev/null', fcntl.O_WRONLY)
-		unistd.dup2(null, unistd.STDOUT_FILENO)
-		unistd.dup2(null, unistd.STDERR_FILENO)
-		if null > 2 then
-			unistd.close(null)
-		end
-
-		unistd.dup2(inr, unistd.STDIN_FILENO)
-		unistd.close(inr)
-
-		unistd.execp('passwd', {[0] = 'passwd'})
-		os.exit(127)
-	end
+	local options = {stdin = sp.PIPE,
+			 stdout = sp.DEVNULL,
+			 stderr = sp.DEVNULL}
 
-	unistd.close(inr)
+	local pid, pipe = sp.popen('passwd', {[0] = 'passwd'}, options)
```

Comment:
Is element 0 necessary when it is the same as the command?

Reply:
I thought so, but it is not:

https://github.com/luaposix/luaposix/blob/master/ext/posix/unistd.c#L302-L307

Reply:
resolved in 47f835e254f2046fe2e57c0dfff21a7737a3f011